frame/uniques/src/migration.rs

```diff
@@ -0,0 +1,56 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2017-2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Various pieces of common functionality.
+use super::*;
+use frame_support::{
+	traits::{Get, GetStorageVersion, PalletInfoAccess, StorageVersion},
+	weights::Weight,
+};
+
+pub fn migrate_to_v1<T: Config<I>, I: 'static, P: GetStorageVersion + PalletInfoAccess>(
+) -> frame_support::weights::Weight {
+	let on_chain_storage_version = <P as GetStorageVersion>::on_chain_storage_version();
+	log::info!(
+		target: "runtime::uniques",
+		"Running migration storage v1 for uniques with storage version {:?}",
+		on_chain_storage_version,
+	);
+
+	if on_chain_storage_version < 1 {
+		let mut count = 0;
+		for (class, detail) in Class::<T, I>::iter() {
+			ClassAccount::<T, I>::insert(&detail.owner, &class, ());
```

Comment:
Note that this migration might be very big and cause issues on a parachain.

Reply:
What is the best practice to bound the migrations. do we just set an upper limit?!